src/main/java/frc/robot/hardware/mechanisms/FlywheelSimulation.java

```diff
@@ -0,0 +1,47 @@
+package frc.robot.hardware.mechanisms;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.simulation.FlywheelSim;
+import frc.utils.time.TimeUtils;
+
+public class FlywheelSimulation extends MechanismSimulation {
+
+	private final FlywheelSim flywheelSimulation;
+
+	private Rotation2d position;
+
+	private Rotation2d currentVelocity;
+
+	public FlywheelSimulation(FlywheelSim flywheelSimulation, double gearRatio) {
+		super(gearRatio);
+		this.flywheelSimulation = flywheelSimulation;
+		this.position = Rotation2d.fromDegrees(0);
```

Comment:
DIP

Reply:
You cant get the position, the simulation starts at 0 and the simulation doesn't have position at all we create it